openthread-tock/platform/platform-tock/radio.c

```diff
@@ -0,0 +1,281 @@
+// TODO COPYRIGHT
+
+#include <ieee802154.h>
+#include <openthread-system.h>
+#include <openthread/platform/radio.h>
+#include <plat.h>
+#include <stdio.h>
+
+#define ACK_SIZE 3
+
+
+static uint8_t tx_mPsdu[OT_RADIO_FRAME_MAX_SIZE];
+static otRadioFrame transmitFrame = {
+  .mPsdu   = tx_mPsdu,
+  .mLength = OT_RADIO_FRAME_MAX_SIZE
+};
+
+// nrf52840 ACK on transmit is currently unimplemented. We fake this 
+// for now by always saying the sent packet was ACKed by the receiver.
+static uint8_t ack_mPSdu[ACK_SIZE] = {0x02, 0x00, 0x00};
+static otRadioFrame ackFrame = {
+  .mPsdu   = ack_mPSdu,
+  .mLength = 3
+};
+
+void otPlatRadioGetIeeeEui64(otInstance *aInstance, uint8_t *aIeeeEui64) {
+  // TODO HARDCODED FOR NOW
```

Comment:
I assume this won't work for a network of devices?

Reply:
Long term this isn't great. Interestingly, this looks like it _will work_ for a network of devices since OpenThread generates a somewhat random ipv6 address for the device from this eui64 (said differently, I have not seen the same ipv6 address repeated while testing).

Adding support for obtaining the eui64 is on my todo list for this week.